aip/general/0213/aip.md

```diff
@@ -0,0 +1,214 @@
+<!-- prettier-ignore-end -->
+
+### Protobuf types
+
+The [`google.protobuf`][protobuf] package is somewhat special in that it is
+shipped with protocol buffers itself, rather than with API tooling. The
+Well-Known Types defined in this package should always be used when
+appropriate, and the [AIP common components][] repo **does not** define any
+protos for these types, even when it defines a corresponding JSON Schema. These
+include:
+
+- [`google.protobuf.Duration`][duration]: Durations, with nanosecond-level
+  precision. The protobuf runtime provides helper functions to convert to and
+  from language-native duration objects where applicable (such as Python's
+  [`timedelta`][timedelta]).
+- [`google.protobuf.Timestamp`][timestamp]: Timestamps, with nanosecond-level
+  precision. The protobuf runtime provides helper functions in most languages
+  to convert to and from language-native timestamp objects (such as Python's
+  [`datetime`][datetime]).
+
+`google.protobuf` also provides some useful components that correspond to JSON
+primitives (and so have no representation at all in the [AIP common
+components][] repo), namely:
+
+- [`google.protobuf.Value`][struct]: An arbitrary JSON value. The protobuf
+  runtime provides helper functions in most languages to convert `Value`
+  objects to and from JSON.
+- [`google.protobuf.Struct`][struct]: JSON-like structures (a dictionary of
+  primitives, lists, and other dictionaries). The protobuf runtime provides
+  helper functions in most languages to convert `Struct` objects to and from
+  JSON.
+
+`google.protobuf.Struct` and `google.protobuf.Value` are designated common
+components by this AIP; proto-based APIs **should** use them when appropriate.
+
+<!-- prettier-ignore-start -->
+[datetime]: https://docs.python.org/3/library/datetime.html#datetime.datetime
+[duration]: https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/duration.proto
+[protobuf]: https://github.com/protocolbuffers/protobuf/tree/main/src/google/protobuf
+[struct]: https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/struct.proto
+[timedelta]: https://docs.python.org/3/library/datetime.html#datetime.timedelta
+[timestamp]: https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/timestamp.proto
+<!-- prettier-ignore-end -->
+
+## Libraries for common types
+
+For the common components in the `aip.type` namespace, which represent common
+data types, the [AIP common components][] repo may contain canonical libraries
+in a number of languages. These libraries are designed to be idiomatic in a
+given language, and should feel similar to using the language's standard
+libraries. They should provide basic functionality like adding two `Money`
+values, or determining if one `Date` comes before another.
+
+When a language already has a standard library representation of a common type
+(such as Python's `datetime` for the `Timestamp` type), there may instead be a
```

Comment:
```suggestion
(such as Python's `datetime` for the `Timestamp` type), there **may** instead be a
```

also: instead of what?

Reply:
Instead of a standalone library like the previous paragraph describes...open to suggestions for how to improve the wording.

Reply:
Regarding the suggestion, I didn't mean to use `**may**` here, I was just using it as a standard English word to mean we aren't promising to provide such things.